zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -238,79 +229,67 @@ impl<F: Field> MPTConfig<F> {
         let mult_table: [Column<Advice>; 2] =
             [meta.advice_column(), meta.advice_column_in(SecondPhase)];
 
-        let memory_columns = (0..5).map(|_| meta.advice_column()).collect::<Vec<_>>();
+        let mut cb = MPTConstraintBuilder::new(5, Some(challenges), None);
+
+        meta.create_gate("MPT lookup tables", |meta| {
+            circuit!([meta, cb], {
+                require!(@MptTableType::Keccak => keccak_table.table_exprs(meta));
+                require!(@MptTableType::Byte => vec![fixed_table.table_exprs(meta)[2].clone()]);
+                require!(@MptTableType::Fixed => fixed_table.table_exprs(meta));
+                require!(@MptTableType::Mult => mult_table.table_exprs(meta));
+            });
+            vec![0.expr()]
+        });
```

Comment:
I still advocate for a `proload_table()` API cuz it looks cleaner. Like you said, ideally one `meta.create_gate` is preferred.

Reply:
It is possible to just call `store_table()` directly which I did for a long time, but I'm conflicted because currently `store_table` uses the active condition in the constraint builder, and so I think it's nice that all circuit code that is part of the main program flow is written inside a ` circuit!` to make that clear. But I also do see the point of a function that doesn't do that and indeed just stores the table columns as is in the table without the active condition applied, but then there's two methods to store tables which I wanted to avoid as well. I guess another function `load_table()` that doesn't apply the current condition could be useful and worthwhile.

Reply:
`load_table` LGTM! it should be the standard workflow for fixed table